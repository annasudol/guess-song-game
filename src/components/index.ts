export * from './alert/Alert';
export * from './button/Button';
export * from './button/ButtonLogout';
export * from './image/Image';
export * from './game/Game';
export * from './login/Login';
export * from './listSongs/ListSongs';
export * from './navigation/Navigation';
export * from './errorMessage/ErrorMessage';
export * from './playlists/Playlists';
export * from './playListItem/PlaylistItem';
export * from './scoreInfo/ScoreInfo';
export * from './scores/Scores';
export * from './summary/Summary';
export * from './search/Search';
export * from './song/Song';
export * from './timer/Timer';
