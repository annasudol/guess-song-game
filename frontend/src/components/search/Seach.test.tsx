/* eslint-disable jest/expect-expect */
import * as redux from 'react-redux';
import { Search } from '../../components';
import { act } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

jest.mock('react-redux');

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  useDispatch: (): jest.Mock<any, any> => mockDispatch,
  useSelector: jest.fn(),
}));

describe('component', (): void => {
  describe('Search', (): void => {
    it('match snapshot', (): void => {
      const spy = jest.spyOn(redux, 'useSelector');

      spy.mockReturnValue({
        token:
          'BQAY_Ky0EiDo4z-U-oC2rYg5aF_kcSUhbukZ7BOJkXYQJzI-ChNz0hrULaIAoTmZ6s3_W_8n2LBh6_OKm5vgtvthrWLaPnGwYutjlGFW4cwQKqvN3ovaJmksGuzj5inFTH3UpEJ6_biTyDr_0w52EBirZQLGDN42mtS7ygfv2Nz',
      });
    });

    const wrapper = shallow(<Search />);

    // eslint-disable-next-line jest/no-standalone-expect
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('change input value when calling onChange function', (): void => {
    const wrapper = shallow(<Search />);

    wrapper.find('input').prop<Function>('onChange')({
      target: { value: 'test' },
    });

    expect(wrapper.find('input').prop('value')).toBe('test');
  });

  it('calls handleSubmit function after submit form', (): void => {
    const wrapper = shallow(<Search />);
    const mockPreventDefault = jest.fn();
    const mockEvent = {
      preventDefault: mockPreventDefault,
    };

    act(() => {
      wrapper.find('form').prop<Function>('onSubmit')(mockEvent);
    });

    wrapper.update();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('calls dispatch after change input and submit', (): void => {
    const wrapper = shallow(<Search />);
    const mockPreventDefault = jest.fn();
    const mockEvent = {
      preventDefault: mockPreventDefault,
    };

    act(() => {
      wrapper.find('input').prop<Function>('onChange')({
        target: { value: 'test' },
      });

      wrapper.find('form').prop<Function>('onSubmit')(mockEvent);
    });

    wrapper.update();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
