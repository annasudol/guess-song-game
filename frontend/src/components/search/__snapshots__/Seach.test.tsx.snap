// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="search ml-1 mr-1 md:flex justify-between items-center"
>
  <h1
    className="font-albaSuper text-center text-orange-500 text-2xl md:text-3xl pb-2 md: pt-2"
  >
    let's choose the playlist
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-sm md:text-xl"
      onChange={[Function]}
      placeholder="Search for playlist"
      value=""
    />
  </form>
</div>
`;
